Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp

```diff
@@ -660,6 +667,37 @@ MediaPlayerEnums::SupportsType GStreamerRegistryScanner::isContentTypeSupported(
 
     const auto& codecs = contentType.codecs();
 
+#if ENABLE(ENCRYPTED_MEDIA)
+    String cryptoblockformat = contentType.parameter("cryptoblockformat"_s);
+    if (!cryptoblockformat.isEmpty()) {
+        auto factories = ElementFactories({ ElementFactories::Type::Decryptor });
+
+        // https://developers.google.com/youtube/devices/living-room/certification-requirements
+        // section 15.1.3: The isTypeSupported method MUST support the cryptoblockformat mime-type
+        // parameter, and MUST support subsample as a parameter value.
+        if (cryptoblockformat != "subsample"_s)
+            return SupportsType::IsNotSupported;
+
+#if GST_CHECK_VERSION(1, 22, 0)
+        for (const auto& mimeCodec : codecs) {
+            auto codecCaps = adoptGRef(gst_codec_utils_caps_from_mime_codec(mimeCodec.ascii().data()));
+            if (!codecCaps) {
+                GST_WARNING("Unable to convert codec %s to caps", mimeCodec.ascii().data());
+                continue;
+            }
+            auto* structure = gst_caps_get_structure(codecCaps.get(), 0);
+            const char* name = gst_structure_get_name(structure);
+            auto caps = adoptGRef(gst_caps_new_simple("application/x-cenc", "original-media-type", G_TYPE_STRING, name, nullptr));
+            if (!factories.hasElementForCaps(ElementFactories::Type::Decryptor, caps))
+                return SupportsType::IsNotSupported;
+        }
+#else
+        if (!factories.hasElementForMediaType(ElementFactories::Type::Decryptor, "application/x-cenc"))
```

Comment:
Shouldn't we also check for `application/x-webm-enc` as well?

Reply:
I thought about yes, but wasn't sure what to do... When is that format selected? For matroska containers only?

Reply:
Yes, WebM, Widevine only.

Reply:
The problem is, when checking for something like `isTypeSupported('video/webm; codecs="vp09..."; cryptoblockformat=subsample')` should we return IsNotSupported when no webm-enc decryptor was found? or cenc? or neither of those? Not sure which semantics to use here.

Reply:
IMHO, we should follow what [Chromium does](https://cobalt.googlesource.com/cobalt/+/refs/heads/20.lts.stable/src/starboard/shared/starboard/media/media_util.cc#138) and support only `cryptoblockformat=subsample` and only when used in webm and the WebM decryptor is present.

Reply:
Sure but you can have vp8/vp9 encrypted in isomp4 too

Reply:
Yes, but apparently Chromium supports is only for WebM so I'd keep it for that only and return `IsNotSupported` for the rest.